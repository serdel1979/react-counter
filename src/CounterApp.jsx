import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {


    const [counter, setCounter] = useState(value)

    const handleAdd = ()=>{
       // setCounter( counter + 1);
       //otra opcion
       setCounter((c)=> c + 1);
    }

    const handleSub = ()=>{
        setCounter((c)=> c - 1);
    }

    const handleReset = ()=>{
        setCounter((c)=> c = value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }>
                +1
            </button>
            <button onClick={ handleSub }>
                -1
            </button>
            <button aria-label='btn-reset' onClick={ handleReset }>
                Reset
            </button>
        </>
    )
}


CounterApp.propTypes={
    value: PropTypes.number.isRequired
}
