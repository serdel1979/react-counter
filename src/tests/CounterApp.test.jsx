const { render, screen } = require("@testing-library/react")
import {CounterApp} from "../CounterApp";

describe('Prueba del counterApp', () => {

    const value = 20;
    test('Debe hacer match con el snapshot ', () => {
        const { container } = render(<CounterApp value={value}/>)
        expect( container ).toMatchSnapshot();
    })
    

    test('Debe mostrar el valor inicial de 20', () => {
        const { container } = render(<CounterApp value={value}/>)
        expect( container ).toMatchSnapshot();
        expect( screen.getByText(value)).toBeTruthy();
    })
  
})
