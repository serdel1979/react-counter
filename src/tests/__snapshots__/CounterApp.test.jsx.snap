// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del counterApp Debe hacer match con el snapshot  1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    20
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;

exports[`Prueba del counterApp Debe mostrar el valor inicial de 20 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    20
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
